### DISCUSSION TITLE
observed property names / controlled vocabulary

### DISCUSSION THREAD

**Discussion by david-i-berry** [wmo-im/tt-cdm#16] [Q&A]
As part of the data model definition we need a way to specify the observed property, e.g. air_temperature etc. From an existing practices point of view what is currently used, e.g. names from the WMO code tables (e.g. BUFR and earlier TAC), WIGOS metadata representation, CF standard_names, other?

What is used for the observed properties in:
- SURFACE
- Climsoft
- CliDE
- MCH
- others

?

**Comment by isedwards** [comment]
@david-i-berry in Climsoft the administrator can customise the software and use their preferred method to specify the observed property. However `climsoft4_element_codes.xls` ([here](https://github.com/opencdms/opencdms/tree/dffde8362d9154dd2291a0350a7049c33a89d192/opencdms/db/schemas/climsoft/v4/docs)) defines the default codes that are generally used.

I'm currently trying to represent `maximum daily temperature (TMPMAX)` and `minimum daily temperature (TMPMIN)` using v0.1.0 of the Climate Data Model.

We've recently discussed using [`phenomenon_start` and `phenomenon_end`](https://github.com/opencdms/opencdms/blob/main/opencdms/db/schemas/wmo-cdm/wmo-cdm-0.0.1.sql#L348-L354) (and other alternatives for specifying the 'daily' range/period).

However, I'm now stuck on the representation of `maxiumum` and `minimum` using CF standard names in the current version of the [observed_property](https://github.com/opencdms/opencdms-test-data/blob/bf710060ec328f95143ee0b2d33c4a34ba273360/data/cdm/code_tables/observed_property.csv) table.

My recollection is that I should use `air_temperature` with an additional "cell method" to specify `"time: maximum"` or `"time: minimum"` ([see 7.3 Cell Methods](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/ch07s03.html)).

Any thoughts on how we should represent this in the `observed_property` table? Cell methods can get quite elaborate but it could be useful to support them? There are Python libraries that can help us to parse and interpret cell methods (e.g. [iris.fileformats.netcdf.parse_cell_methods](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris.fileformats.netcdf.html#iris.fileformats.netcdf.parse_cell_methods)).
